backend/service/xkcd.go

```diff
@@ -0,0 +1,88 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+	"xkcd/comics/model"
+)
+
+const xkcdURL = "https://xkcd.com/%d/info.0.json"
+
+func GetComic(comicNum int) (*model.Comic, error) {
+    resp, err := http.Get(fmt.Sprintf(xkcdURL, comicNum))
+    if err != nil {
+        return nil, err
```

Comment:
We want to extract as much information as possible from errors, such that when we read the logs, it's easy to figure out where the issue is coming from. Simply returning the error may make this difficult.

For instance, if this function returns an error, you wouldn't know if it was from the HTTP request or from the JSON unmarshalling.

You can wrap your error with ```fmt.Errorf``` to provide more info, such as:
```go
return nil, fmt.Errorf("failed to GET comic %d: %s", comicNum, err)
```

Reply:
Please apply this feedback to all places where you return errors.